ui/component/livestreamChatLayout/view.jsx

```diff
@@ -0,0 +1,375 @@
+};
+
+export default function LivestreamChatLayout(props: Props) {
+  const {
+    claim,
+    comments: commentsByChronologicalOrder,
+    embed,
+    isPopoutWindow,
+    pinnedComments,
+    superChats: superChatsByAmount,
+    uri,
+    doCommentList,
+    doCommentSocketConnect,
+    doCommentSocketDisconnect,
+    doResolveUris,
+    doSuperChatList,
+  } = props;
+
+  const {
+    location: { pathname },
+  } = useHistory();
+
+  const isMobile = useIsMobile();
+
+  const discussionElement = document.querySelector('.livestream__comments');
+
+  const restoreScrollPos = React.useCallback(() => {
+    if (discussionElement) discussionElement.scrollTop = 0;
+  }, [discussionElement]);
+
+  const commentsRef = React.createRef();
+
+  const [viewMode, setViewMode] = React.useState(VIEW_MODES.CHAT);
+  const [scrollPos, setScrollPos] = React.useState(0);
+  const [showPinned, setShowPinned] = React.useState(true);
+  const [resolvingSuperChats, setResolvingSuperChats] = React.useState(false);
+  const [mention, setMention] = React.useState();
+  const [openedPopoutWindow, setPopoutWindow] = React.useState(false);
+  const [chatHidden, setChatHidden] = React.useState(false);
+
+  const quickMention =
+    mention && formatLbryUrlForWeb(mention).substring(1, formatLbryUrlForWeb(mention).indexOf(':') + 3);
+  const claimId = claim && claim.claim_id;
+  const commentsToDisplay = viewMode === VIEW_MODES.CHAT ? commentsByChronologicalOrder : superChatsByAmount;
+  const commentsLength = commentsToDisplay && commentsToDisplay.length;
+  const pinnedComment = pinnedComments.length > 0 ? pinnedComments[0] : null;
+
+  let superChatsChannelUrls = [];
+  let superChatsFiatAmount = 0;
+  let superChatsLBCAmount = 0;
+  if (superChatsByAmount) {
+    superChatsByAmount.forEach((superChat) => {
+      const { is_fiat: isFiat, support_amount: tipAmount, channel_url: uri } = superChat;
+
+      if (isFiat) {
+        superChatsFiatAmount = superChatsFiatAmount + tipAmount;
+      } else {
+        superChatsLBCAmount = superChatsLBCAmount + tipAmount;
+      }
+      superChatsChannelUrls.push(uri || '0');
+    });
+  }
+
+  function toggleSuperChat() {
+    if (superChatsChannelUrls && superChatsChannelUrls.length > 0) {
+      doResolveUris(superChatsChannelUrls);
+
+      if (superChatsByAmount.length > LARGE_SUPER_CHAT_LIST_THRESHOLD) {
+        setResolvingSuperChats(true);
+      }
+    }
+    setViewMode(VIEW_MODES.SUPERCHAT);
+  }
+
+  function handlePopout() {
+    const newWindow = window.open('/$/popout' + pathname, 'Popout Chat', 'height=700,width=400');
+
+    // Add function to newWindow when closed manually
+    // show the chat again and re-connect to websocket
+    newWindow.onbeforeunload = () => {
+      if (claimId) doCommentSocketConnect(claimId);
+      setPopoutWindow(undefined);
+    };
+
+    if (window.focus) newWindow.focus();
+    setPopoutWindow(newWindow);
+
+    // Disconnect since it will re-connect on the new window
```

Comment:
If you press F5 on the popup, the main window will repopulate the chat, and also reconnect websocket.  We have 2 instances.  Ok with that?

![image](https://user-images.githubusercontent.com/64950861/149284343-87a53896-dc67-4d27-a74a-f64df63764b0.png)

Reply:
Same goes for this scenario:  pop out, then resize the main window into mobile layout.

Reply:
After having 2 instances, close the popup and you get this:

![image](https://user-images.githubusercontent.com/64950861/149284966-f0dffa95-9057-4a88-9d07-8d7bdfa77c4d.png)

Reply:
Good catch, didn't think of the refresh scenario here. I guess the main tab thinks it's closed, and reopens there. Hmm...probably won't happen too often. Still should be better than double counting (see [previous comment) ](https://github.com/OdyseeTeam/odysee-frontend/pull/681#discussion_r784057254))